src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -39,6 +46,7 @@ export class UserD2ApiRepository implements UserRepository {
                 pageSize,
                 query: search !== "" ? search : undefined,
```

Comment:
The value of `rootJunction` does not seem to affect how the query+filter intersection is being applied (it's always an AND). 

So if we really want to support OR when both kind of filters are enabled, we'll need to do some programmatic filtering. That's not trivial, as now the `pager` from the API is wrong, and AFICS we don't have any way to calculate it other than: getting all the users+ apply programmatic filter + build the pager.

Reply:
My understanding is that we want the AND/OR to be applied only to the advanced filters, not to the query + advanced filters intersection. But I'm not sure about it, the request came via @ifoche (and I can't check the original request mentioned in https://app.clickup.com/t/22k4dbc), so I'll ask him.